### ISSUE TITLE
Add Binder configuration

### ISSUE THREAD

**Issue by lbianchi-lbl** [watertap-org/watertap#1157]
Binder (https://mybinder.org) allows to run Jupyter notebooks from a publicly hosted Git repository in a temporary cloud environment, fully online, free of charge.

To try out how WaterTAP tutorials work on Binder, it should be enough to click on this button: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/lbianchi-lbl/watertap/binder?labpath=tutorials%2Fintroduction.ipynb).

- All the dependencies and prerequisites (including `idaes get-extensions`, system libraries) should be taken care of, so users should be able to just run the starting notebook (or any other notebook accessible through the file browser on the left)
- The button/link above currently points to [`lbianchi-lbl/watertap:binder`](https://github.com/lbianchi-lbl/watertap/tree/binder) since the required config files are not merged in here yet; eventually this would become `watertap-org/watertap:main`, but any other fork/branch combination would work by adapting the launch URL appropriately

**Comment by adam-a-a** [comment]
I am excited about this! One question- is there any way to enforce a custom order for the notebooks in the file browser? For example, we'd want users to start with something simple like how to setup an RO model before delving into other things like setting up the NF model (nawi_spring_meeting notebook).

**Comment by lbianchi-lbl** [comment]
> I am excited about this! One question- is there any way to enforce a custom order for the notebooks in the file browser? For example, we'd want users to start with something simple like how to setup an RO model before delving into other things like setting up the NF model (nawi_spring_meeting notebook).

Good question. The simplest (and widely used, in my experience) method would be to rename the `.ipynb` files by adding leading numbers corresponding to the suggested order, e.g. `01_introduction.ipynb`, `02_ro_model.ipynb`, etc. I can imagine that some of the current or future notebooks might not have an intrinsic order (i.e. they could be run in any order after the introduction), but maybe just assigning numbers arbitrarily would be OK?

**Comment by adam-a-a** [comment]
> > I am excited about this! One question- is there any way to enforce a custom order for the notebooks in the file browser? For example, we'd want users to start with something simple like how to setup an RO model before delving into other things like setting up the NF model (nawi_spring_meeting notebook).
> 
> Good question. The simplest (and widely used, in my experience) method would be to rename the `.ipynb` files by adding leading numbers corresponding to the suggested order, e.g. `01_introduction.ipynb`, `02_ro_model.ipynb`, etc. I can imagine that some of the current or future notebooks might not have an intrinsic order (i.e. they could be run in any order after the introduction), but maybe just assigning numbers arbitrarily would be OK?

@lbianchi-lbl  That sounds fine to me. One other thought--is there an easy way of using Binder to review PRs that add/modify notebooks? I was thinking of something similar to the readthedocs check, where you can click on "Details" and be able to render the documentation (but I don't want to make life difficult). I'm curious to hear your thoughts.

**Comment by lbianchi-lbl** [comment]
> @lbianchi-lbl That sounds fine to me. One other thought--is there an easy way of using Binder to review PRs that add/modify notebooks? I was thinking of something similar to the readthedocs check, where you can click on "Details" and be able to render the documentation (but I don't want to make life difficult). I'm curious to hear your thoughts.

I think that's a great idea and I agree it'd be very useful to have. My initial thoughts on how this could be done:

- The Binder URL follows a predetermined template, so it should be possible to generate it automatically based on the PR number (or the PR head branch)
- At that point, we'd need to make the proper link accessible on the PR interface. One way to do so is editing the description (initial comment) using GHA, similarly to how it's done in `IDAES/examples`: 
![image](https://github.com/watertap-org/watertap/assets/48035537/268fd955-a8f6-4512-9ec5-0b24165dba19)
- This solution uses an Action developed by ReadTheDocs, but my estimate is that it shouldn't be too hard to adapt it to Binder instead (or maybe it already exists)

In summary: I think this is worth doing; and, it should take some, but not a lot of, work to set it up.
